src/nisarqa/processing/insar/quiver_plots.py

```diff
@@ -382,4 +683,261 @@ def truncate_colormap(
+        'time_end':
+           End of search interval, in seconds. Defaults to `orbit.end_time`.
+        Defaults to None.
+    """
+
+    orbit: isce3.core.Orbit
+    wavelength: float
+    look_side: isce3.core.LookSide | str
+    geo2rdr_params: Mapping[str, float] | None = None
+
+
+def get_offset_values_in_projected_coordinates(
+    az_offsets: np.ndarray,
+    rg_offsets: np.ndarray,
+    geo_grid: nisarqa.GeoGrid,
+    projection_params: nisarqa.ParamsForAzRgOffsetsToProjected,
+) -> tuple[np.ndarray, np.ndarray]:
+    """
+    Convert pixel values from offset in az/range to offset in projected X/Y.
+
+    For the along-track and slant range offsets raster in NISAR GOFF and GUNW
+    products, the raster images are geocoded to projected coordinates but
+    their pixel values represent offsets in azimuth and slant range directions
+    (respectively) in meters.
+    This function will convert the values from offset amount in azimuth
+    and slant range directions to the offset in the X and Y directions
+    of the projected image grid.
+
+    Parameters
+    ----------
+    rg_offsets : numpy.ndarray
+        2D array of the values of the slant range offsets.
+        Array should be geocoded, but the pixel values should represent
+        offset in slant range direction (in meters).
+    az_offsets : numpy.ndarray
+        2D array of the values of the along track offsets.
+        Array should be geocoded, but the pixel values should represent
+        offset in along azimuth direction (in meters).
+    geo_grid : nisarqa.GeoGrid
+        GeoGrid parameters for `rg_offsets` and `az_offsets`.
+    projection_params : ParamsForAzRgOffsetsToProjected
+        Parameters used to convert the offsets rasters' pixel values from
+        azimuth/range offset values (in meters) to values in the
+        rasters' projected coordinate grid.
+
+    Returns
+    -------
+    offset_in_x_direction, offset_in_y_direction : numpy.ndarray
+        2D array of the offset values in X and Y direction (respectively)
+        of the projected coordinate grid.
+    """
+
+    # Variable renaming and unpacking
+    rg_offsets_at_arrow_tails = rg_offsets
+    az_offsets_at_arrow_tails = az_offsets
+    epsg = geo_grid.epsg
+    x_coords_values_at_arrow_tails = geo_grid.x_coordinates
+    y_coords_values_at_arrow_tails = geo_grid.y_coordinates
+    orbit = projection_params.orbit
+    wavelength = projection_params.wavelength
+    look_side = projection_params.look_side
+    geo2rdr_params = projection_params.geo2rdr_params
+    geo2rdr_params = {} if (geo2rdr_params is None) else geo2rdr_params
+
+    # Useful variables for the algorithm
+    num_arrows_x = len(x_coords_values_at_arrow_tails)
+    num_arrows_y = len(y_coords_values_at_arrow_tails)
+
+    if (rg_offsets_at_arrow_tails.shape[1] != num_arrows_x) or (
+        rg_offsets_at_arrow_tails.shape[0] != num_arrows_y
+    ):
```

Comment:
What about the azimuth offsets? Do we need to check that their shape is consistent?